torch/utils/checkpoint.py

```diff
@@ -286,13 +295,13 @@ def backward(ctx, *args):
             if ctx.preserve_rng_state:
                 torch.set_rng_state(ctx.fwd_cpu_state)
                 if ctx.had_device_in_fwd:
-                    set_device_states(ctx.fwd_devices, ctx.fwd_device_states)
+                    set_device_states(ctx.fwd_devices, ctx.fwd_device_states, device_type=ctx.device)
```

Comment:
This is a preexisting issue, but do you mind renaming `ctx.device` and `device` to be `device_type`. It confused me when was reading this.

Reply:
Will do.

Reply:
renamed `device` to `device_type`